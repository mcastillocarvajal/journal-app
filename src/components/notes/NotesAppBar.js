import React from 'react'

export const NotesAppBar = () => {
    return (
        <div className="notes__appbar">
            <span>January 1, 2012</span>

            <div>

                <button className="btn">
                    Picture
                </button>

                <button className="btn">
                    Save
                </button>

            </div>
        </div>
    )
}
