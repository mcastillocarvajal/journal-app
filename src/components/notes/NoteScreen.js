import React, { useEffect, useRef } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { activeNote } from '../../actions/notes';
import { useForm } from '../../hooks/useForm';
import { NotesAppBar } from './NotesAppBar'


export const NoteScreen = () => {

    const dispatch = useDispatch();

    const { active:note } = useSelector(state => state.notes);

    const [ formValues, handleInputChange, reset ] = useForm( note );

    const { body, title } = formValues;

    const activeId = useRef( note.id )

    useEffect(() => {

        if ( note.id !== activeId.current ) {
            reset( note );
            activeId.current = note.id
        }

    }, [ note, reset ]);

    useEffect(() => {

        dispatch( activeNote( formValues.id, { ...formValues} ) );

    }, [ formValues, dispatch ])


    return (
        <div className="notes__main-content">

            <NotesAppBar />

            <div className="notes__content">

                <input  
                    type="text"
                    className="notes__title-input"
                    placeholder="Some awesome words"
                    autoComplete="off"
                    name='title'
                    value={ title }
                    onChange={ handleInputChange }
                />

                <textarea 
                    placeholder="What happened today?"
                    className="notes__textarea"
                    name='body'
                    value={ body }
                    onChange={ handleInputChange }
                ></textarea>
            </div>

            {
                (note.url) &&
                <div className="notes-image">

                    <img
                        src="https://www.businessinsider.in/photo/81769906/How-to-reverse-image-search-on-Google-to-find-information-related-to-a-specific-photo.jpg?imgsize=297676"
                        alt="img"
                    />
                </div>
            }

        </div>
    )
}
