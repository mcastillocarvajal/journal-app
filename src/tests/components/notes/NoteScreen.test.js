import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { activeNote } from '../../../actions/notes';
import { NoteScreen } from '../../../components/notes/NoteScreen';

jest.mock('../../../actions/notes', () => ({
    activeNote: jest.fn()
}))

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {
        uid: '1',
        name: 'Moises'
    },
    ui: {
        loading: false,
        msgError: null
    },
    notes: {
        active: {
            id: '1234',
            date: 2321445,
            title: 'title',
            url: 'http://nkanklf.com',
            body: 'body'
        },
        notes: []
    }
};

let store = mockStore( initState );
store.dispatch = jest.fn();

const wrapper = mount( 
    <Provider store={ store }>
            <NoteScreen />
    </Provider>
);

describe('test for <NoteScreen />', () => {
    
    test('should match snapshot', () => {
        
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('should trigger the activeNote', () => {
        
        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title',
                value: 'Hello test'
            }
        })

        expect( activeNote ).toHaveBeenLastCalledWith(
            '1234',
            {
                date: 2321445,
                id: '1234',
                title: 'Hello test',
                url: 'http://nkanklf.com',
                body: 'body'
            }
        );
    })
    
})
