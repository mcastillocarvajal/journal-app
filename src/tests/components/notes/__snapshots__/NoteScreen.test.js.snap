// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for <NoteScreen /> should match snapshot 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      Wednesday, December 31st 1969.
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome words"
      type="text"
      value="title"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happened today?"
      value="body"
    />
  </div>
  <div
    className="notes-image"
  >
    <img
      alt="img"
      src="http://nkanklf.com"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
